src/transformers/pipelines/text_to_speech.py

```diff
@@ -0,0 +1,173 @@
+        else:
+            self.sampling_rate = sampling_rate
+
+    def preprocess(self, text, speaker_embeddings=None, **kwargs):
+        if self.is_speecht5:
+            inputs = self.processor(text=text, return_tensors="pt")
+            if speaker_embeddings is None:
+                embeddings_dataset = load_dataset("Matthijs/cmu-arctic-xvectors", split="validation")
+                speaker_embeddings = torch.tensor(embeddings_dataset[7305]["xvector"]).unsqueeze(0)
+
+            return {"input_ids": inputs["input_ids"], "speaker_embeddings": speaker_embeddings}
+
+        processor_args_dict = {}
+
+        SPEAKER_EMBEDDINGS_KEY_MAPPING.get(self.model_type, "speaker_embeddings")
+        if self.model.config.model_type == "bark":
+            # bark speaker embeddings is passed as voice_preset in its processor
+            processor_args_dict["voice_preset"] = speaker_embeddings
+        else:
+            processor_args_dict["speaker_embeddings"] = speaker_embeddings
+
+        output = self.processor(text, **processor_args_dict, **kwargs)
+
+        return output
+
+    def _forward(self, model_inputs, **kwargs):
+        if self.is_speecht5:
+            inputs = model_inputs["input_ids"]
+
+            speaker_embeddings = model_inputs["speaker_embeddings"]
+
+            with torch.no_grad():
+                speech = self.model.generate_speech(inputs, speaker_embeddings, vocoder=self.vocoder)
+        else:
+            if self.only_one_speaker_embeddings:
+                speaker_embeddings_key = SPEAKER_EMBEDDINGS_KEY_MAPPING.get(self.model_type, "speaker_embeddings")
+
+                # check batch_size > 1
+                if len(model_inputs["input_ids"]) > 1 and model_inputs.get(speaker_embeddings_key, None) is not None:
+                    model_inputs[speaker_embeddings_key] = model_inputs[speaker_embeddings_key][0]
+
+            with torch.no_grad():
+                speech = self.model.generate(**model_inputs, **kwargs)
+
+        return speech
+
+    def __call__(
+        self,
+        input_texts: Union[str, List[str]],
+        **generate_kwargs,
+    ):
+        """
+        Generates speech/audio from the inputs. See the [`TextToSpeechPipeline`] documentation for more information.
+
+        Args:
+            input_texts (`str` or `List[str]`):
+                The text(s) to generate.
+            speaker_embeddings (`str` or `torch.Tensor` or `Dict[np.ndarray]`, *optional*):
+                The speaker prompt, i.e the speaker embeddings conditionning the inputs.
+            generate_kwargs (*optional*):
+                Remaining parameters passed to the model generation method.
+
+        Return:
+            A `torch.Tensor` or a list of `torch.Tensor`: Each result comes as a `torch.Tensor` corresponding to the
+            generated audio.
+        """
+        return super().__call__(input_texts, **generate_kwargs)
+
+    def _sanitize_parameters(
+        self,
+        **generate_kwargs,
```

Comment:
Do no pass generate_kwargs as is.

We're having lots of issues on other pipeliens that did that.

Either whitelist the subset you want to allow or send a raw dict

`pipeline(..., generate_kwargs={...}` 

The issue with doing this is that you're not knowing what params are or are not used, and at any point `generate` might add support for new kwargs that could clash with some in the pipeline.

This is already the case with `max_length` which exists for `generate` but also for the tokenization and have different meaning.